src/_guides/language/sound-dart.md

```diff
@@ -0,0 +1,696 @@
+---
+layout: guide
+title: "Sound Dart Guide"
+description: "."
+---
+
+This guide tells you why and how to write sound (type safe) Dart code.
+You'll learn how to use strong mode to enable soundness, as well as
+how to substitute types safely.
+
+{% comment %}
+Not ready yet...
+If you are already using strong mode Dart, you might also check out
+[Sound Dart: Common Problems](/guides/language/sound-dart-common-problems).
+{% endcomment %}
+
+<aside class="alert alert-info" markdown="1">
+**Note:** The terms "sound Dart", "strong mode Dart", and "type safe Dart"
+are sometimes used interchangeably. _Strong mode_ is Dart's implementation
+of a sound type system. With strong mode enabled,
+Dart is a type safe language. "Classic Dart" refers to Dart
+before soundness was added to the language.
+
+Eventually, soundness won't be optional in the Dart language, and strong mode
+will no longer be a mode. It will just be part of the language.
+</aside>
+
+By writing sound Dart code today, you'll reap some benefits now,
+with more in the near future. Current benefits include finding bugs
+at compile time (rather than at runtime) using Dart's static analyzer.
+And soon you'll be able to use new tools that quickly and incrementally
+compile your sound Dart code, giving you a better overall
+developer experience.
+
+Sound Dart adds only a few additional rules beyond that for classic
+Dart&mdash;mostly you clarify code where the types are ambiguous or
+incorrect. In fact, most strong mode errors can be fixed by adding type
+annotations to your Lists and Maps. The following code shows valid
+but unsound Dart code. The `fn()` function prints an integer list.
+The `main()` function creates a list of integers and passes it to `fn()`:
+
+<div class="fails-sa" markdown="1">
+{% prettify dart %}
+void fn(List<int> a) {
+  print(a);
+}
+
+main() {
+  var list = [];
+  list.add(1);
+  list.add(2);
```

Comment:
I think this example would be a better introduction to soundness if the code was something that worked, but that also didn't really match the documented intent.  So for example, using `list.add("2")`.